### PR TITLE
Bug 1225330 - Status bar icons change color before homescreen transition when home button is pressed

### PR DESCRIPTION
**Pull request by jedireza**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ee031a517837029bcffe374e30086cd0bbebb471

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=189537063adca3f66705cc9ba9b3afb194daad65

File: apps/system/test/unit/update_manager_test.js [hunk 1]
```diff
@@ -428,8 +428,6 @@ suite('system/UpdateManager', function() {
         assert.isFalse(UpdateManager.downloadViaDataConnectionDialog.
           classList.contains('visible'));
         assert.isFalse(MockCustomDialog.mShown);
-        // should not close all custom dialog when no custom dialog is opened
-        assert.isFalse(Service.request.calledWith('hideCustomDialog'));
```
Comment (jedireza) [inline apps/system/test/unit/update_manager_test.js]: @gasolin I know it was a while ago (July) that you added this.

All the other show dialog functions were setting `UpdateManager._hasDialog` to `true` except `UpdateManager.showDownloadPrompt()`. In this PR I do the same in `UpdateManager.showDownloadPrompt()`. 

Now, when we handle the `home` and `holdhome` events, we know to only call `UpdateManager.cancelPrompt()` if there is a dialog showing.

Do you have any thoughts or recommendations?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=63c84bd76777c438cf71e9c8a5257fa3c005b872

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=58ab05f87b0843d9634ebcf99742591d0c06165c

**Pull request merged**